source/extensions/access_loggers/wasm/config.cc

```diff
@@ -20,30 +20,29 @@ namespace AccessLoggers {
 namespace Wasm {
 
 AccessLog::InstanceSharedPtr
-WasmAccessLogFactory::createAccessLogInstance(const Protobuf::Message& config,
+WasmAccessLogFactory::createAccessLogInstance(const Protobuf::Message& proto_config,
                                               AccessLog::FilterPtr&& filter,
                                               Server::Configuration::FactoryContext& context) {
-  const auto& wal_config =
-      MessageUtil::downcastAndValidate<const envoy::config::accesslog::v2::WasmAccessLog&>(config);
+  const auto& config =
+      MessageUtil::downcastAndValidate<const envoy::config::accesslog::v2::WasmAccessLog&>(
+          proto_config);
+  auto id = config.id();
+  auto configuration = std::make_shared<std::string>(config.configuration());
   auto tls_slot = context.threadLocal().allocateSlot();
-  if (wal_config.has_vm_config()) {
-    auto base_wasm =
-        Common::Wasm::createWasm(wal_config.id(), wal_config.vm_config(), context.api());
-    tls_slot->set([&wal_config, &base_wasm, &context](Event::Dispatcher& dispatcher) {
-      auto result =
-          Common::Wasm::createThreadLocalWasm(*base_wasm, wal_config.vm_config(), dispatcher,
-                                              wal_config.configuration(), context.api());
-      result->setClusterManager(context.clusterManager());
+  if (config.has_vm_config()) {
+    auto base_wasm = Common::Wasm::createWasm(id, config.vm_config(), context.clusterManager(),
```

Comment:
Currently the base_wasm is used to verify that the code loads so that the exception will appear in the context of the Filter/Service creation.  We could separate this out and give it a new name, but perhaps the same result could be adopted with comments?  I'll add some and then we can decide what we want to do as follow on work.

Reply:
Erm, my point was that if we instantiate a VM that we don't need, then we're going to be allocating a lot of memory for a VM that's never used (esp. in case of modules with module-managed vs runtime-managed memory).

But yeah, let's discuss this and do as a follow-up PR. I've already added a TODO for it in the tracker, so we probably don't even need extra comments here.

Reply:
SGTM